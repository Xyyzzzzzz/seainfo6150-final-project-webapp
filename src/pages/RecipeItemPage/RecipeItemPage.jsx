import React, { useState } from 'react'
import styles from './RecipeItemPage.module.css'

const RecipeItemPage = ({ recipe, myCollection, setMyCollection }) => {
    const recipeId = recipe.id;
    const [buttonText, setButtonText] = useState(
        myCollection[recipeId] ? "Remove from My Collection" : "Add to My Collection");

    const ToggleRecipeToCollection = () => {
        const newMyCollection = myCollection;
        if (myCollection[recipeId]) {
            delete newMyCollection[recipeId];
            setButtonText("Add to My Collection");
        } else {
            newMyCollection[recipeId] = recipe;
            setButtonText("Remove from My Collection");
        }
        setMyCollection(newMyCollection);
    }

    return (
        <div className={styles.container}>
            <img className={styles.image} src={recipe.images.recipeDetailUrl} alt={recipe.title}></img>
            <h2>
                {recipe.title}
            </h2>
            <button className="collect-button" onClick={ToggleRecipeToCollection}>
                {buttonText}
            </button>
            <div className={styles.textWrapper}>
                <div className={styles.overview}>
                    <div className={styles.recommendationText}>{recipe.recommendationText}</div>
                    <div className={styles.otherInfo}>
                        <div>
                            Serving Number: {recipe.otherInfo.servingNumber}
                        </div>
                        <div>
                            Cooking Time: {recipe.otherInfo.cookTime}
                        </div>
                        <div>
                            Skill Level: {recipe.otherInfo.skillLevel}
                        </div>
                    </div>
                </div>
                <div className={styles.details}>
                    <div className={styles.ingredients}>
                        <h3>Main Ingredients</h3>
                        <ul>
                            {recipe.mainIngredients.map((ingredient, index) => (
                                <li key={index} className={styles.ingredient}>{ingredient}</li>
                            ))}
                        </ul>
                    </div>
                    <div className={styles.instructions}>
                        <h3>Cooking Instructions</h3>
                        <ol>
                            {recipe.cookingInstructions.map((instruction, index) => (
                                <li key={index} className={styles.instruction}>Step {index + 1}: {instruction}</li>
                            ))}
                        </ol>
                    </div>
                </div>
            </div>
        </div >
    )
};

export default RecipeItemPage;