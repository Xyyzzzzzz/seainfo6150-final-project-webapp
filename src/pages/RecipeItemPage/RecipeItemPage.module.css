.container{
    margin: 0 5% 5% 5%;
}

.image{
    width: 100%;
}

.overview{
    margin-top: 20px;
    display: flex;
    flex-direction: column;
    line-height: 1.5;
}

.otherInfo{
    margin-top: 20px;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
}

.details{
    margin-top: 20px;
    display: flex;
    flex-direction: column;
}

.ingredient{
    margin: 5px 0;
    list-style: square;
}

.instruction{
    margin: 10px 0;
    line-height: 1.5;
}

@media screen and (min-width: 600px) {
    .overview{
        display: grid;
        grid-template-columns: 2fr 1fr;
        column-gap: 10%;
        line-height: 1.5;
    }
    .otherInfo{
        margin-top: 0px;
    }
}

@media screen and (min-width: 800px) {
    .details{
        margin-top: 20px;
        display: grid;
        grid-template-columns: 1fr 2fr;
        column-gap: 5%;
    }
}

