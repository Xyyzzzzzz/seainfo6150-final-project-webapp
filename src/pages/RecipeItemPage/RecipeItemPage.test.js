import React from "react";
import { render } from "@testing-library/react";
import RecipeItemPage from "./RecipeItemPage.jsx";

const recipe = {
    "id": 1,
    "title": "Roasted Turkey with Smoked Paprika",
    "category": "American",
    "recommendationText": "Try this recipe for the perfect roasted turkey, made with a variety of herbs and spices. Perfect for the holiday season, the paprika adds incredible flavor for a slight twist on the classic entrée.",
    "images": {
        "recipeDetailUrl": "https://images.albertsons-media.com/is/image/ABS/0_roasted-turkey-with-smoked-paprika-M2?$recipe-detail-hero-desktop$",
        "recipeCardUrl": "https://images.albertsons-media.com/is/image/ABS/0_roasted-turkey-with-smoked-paprika-M2?$recipe-card$"
    },
    "otherInfo": {
        "servingNumber": 8,
        "cookTime": "2 hour",
        "skillLevel": "Medium"
    },
    "mainIngredients": [
        "2 tablespoons Italian seasoning",
        "2 tablespoons sea salt",
        "1 tablespoon coarsely ground black pepper",
        "1 tablespoon smoked paprika",
        "2 teaspoons garlic powder",
        "2 teaspoons ground mustard",
        "1 whole turkey (12 to 14 pounds), fresh or frozen, thawed",
        "2 ribs celery",
        "1 onion, quartered",
        "1 orange, quartered",
        "2 bay leaves",
        "1 tablespoon olive oil"
    ],
    "cookingInstructions": [
        "Place oven rack in lowest position. Preheat oven to 325°. Place roasting rack in shallow roasting pan. Mix first 6 ingredients in small bowl.",
        "Rinse turkey; pat dry. Place turkey, breast-side up, in prepared pan. Sprinkle 1 tablespoon of the seasoning mixture inside turkey. Stuff with celery, onion, orange and bay leaves.",
        "Brush turkey breast with oil. Spread remaining seasoning mixture over entire surface and under skin of turkey. Add 1/2 cup water to pan; cover loosely with heavy duty foil.",
        "Roast one hour. Remove foil. Roast 2 to 2 ½ hours longer or until internal temperature reaches 165° (175° in thigh), basting occasionally with pan juices.",
        "Remove turkey from oven. Let stand 20 minutes. Transfer to platter or carving board and slice. Reserve pan juices to make gravy or to serve with turkey"
    ]
};

const myCollection = {};


it("renders the Recipe Item page correctly", () => {
    const { container } = render(<RecipeItemPage recipe={recipe} myCollection={myCollection} />);
    expect(container).toMatchSnapshot();
});