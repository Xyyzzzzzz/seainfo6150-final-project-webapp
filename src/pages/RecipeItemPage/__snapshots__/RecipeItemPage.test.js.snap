// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Recipe Item page correctly 1`] = `
<div>
  <div
    class="container"
  >
    <img
      alt="Roasted Turkey with Smoked Paprika"
      class="image"
      src="https://images.albertsons-media.com/is/image/ABS/0_roasted-turkey-with-smoked-paprika-M2?$recipe-detail-hero-desktop$"
    />
    <h2>
      Roasted Turkey with Smoked Paprika
    </h2>
    <button
      class="collect-button"
    >
      Add to My Collection
    </button>
    <div
      class="textWrapper"
    >
      <div
        class="overview"
      >
        <div
          class="recommendationText"
        >
          Try this recipe for the perfect roasted turkey, made with a variety of herbs and spices. Perfect for the holiday season, the paprika adds incredible flavor for a slight twist on the classic entrée.
        </div>
        <div
          class="otherInfo"
        >
          <div>
            Serving Number: 
            8
          </div>
          <div>
            Cooking Time: 
            2 hour
          </div>
          <div>
            Skill Level: 
            Medium
          </div>
        </div>
      </div>
      <div
        class="details"
      >
        <div
          class="ingredients"
        >
          <h3>
            Main Ingredients
          </h3>
          <ul>
            <li
              class="ingredient"
            >
              2 tablespoons Italian seasoning
            </li>
            <li
              class="ingredient"
            >
              2 tablespoons sea salt
            </li>
            <li
              class="ingredient"
            >
              1 tablespoon coarsely ground black pepper
            </li>
            <li
              class="ingredient"
            >
              1 tablespoon smoked paprika
            </li>
            <li
              class="ingredient"
            >
              2 teaspoons garlic powder
            </li>
            <li
              class="ingredient"
            >
              2 teaspoons ground mustard
            </li>
            <li
              class="ingredient"
            >
              1 whole turkey (12 to 14 pounds), fresh or frozen, thawed
            </li>
            <li
              class="ingredient"
            >
              2 ribs celery
            </li>
            <li
              class="ingredient"
            >
              1 onion, quartered
            </li>
            <li
              class="ingredient"
            >
              1 orange, quartered
            </li>
            <li
              class="ingredient"
            >
              2 bay leaves
            </li>
            <li
              class="ingredient"
            >
              1 tablespoon olive oil
            </li>
          </ul>
        </div>
        <div
          class="instructions"
        >
          <h3>
            Cooking Instructions
          </h3>
          <ol>
            <li
              class="instruction"
            >
              Step 
              1
              : 
              Place oven rack in lowest position. Preheat oven to 325°. Place roasting rack in shallow roasting pan. Mix first 6 ingredients in small bowl.
            </li>
            <li
              class="instruction"
            >
              Step 
              2
              : 
              Rinse turkey; pat dry. Place turkey, breast-side up, in prepared pan. Sprinkle 1 tablespoon of the seasoning mixture inside turkey. Stuff with celery, onion, orange and bay leaves.
            </li>
            <li
              class="instruction"
            >
              Step 
              3
              : 
              Brush turkey breast with oil. Spread remaining seasoning mixture over entire surface and under skin of turkey. Add 1/2 cup water to pan; cover loosely with heavy duty foil.
            </li>
            <li
              class="instruction"
            >
              Step 
              4
              : 
              Roast one hour. Remove foil. Roast 2 to 2 ½ hours longer or until internal temperature reaches 165° (175° in thigh), basting occasionally with pan juices.
            </li>
            <li
              class="instruction"
            >
              Step 
              5
              : 
              Remove turkey from oven. Let stand 20 minutes. Transfer to platter or carving board and slice. Reserve pan juices to make gravy or to serve with turkey
            </li>
          </ol>
        </div>
      </div>
    </div>
  </div>
</div>
`;
